### ISSUE TITLE
Break BB42 and BB24 monoliths

### ISSUE THREAD

**Issue by tcosmo** [ccz181078/Coq-BB5#10]
`BB42Theorem.v` and `BB24Theorem.v` are currently monoliths that respectively contain the proofs of `BB(4) = 107` and `BB(2,4) = 3,932,964`. These proofs mainly use the same abstractions as BB5 and could re-use the new files created in [the refactor](https://github.com/ccz181078/Coq-BB5/pull/6) of the BB5 proof in multiple files. 

An obstacle is the hard-coding of state and symbol sets which are specifically linked to 5 states and 2 symbols in some of the re-usable parts of the BB(5) proof. Making these generic is considered as "not very easy" by @ccz181078.

I'm pausing my effort on that front but it would be nice, for the elegance and to promote re-usability of Coq-BB5's core components (TNF, Deciders, ListTape, etc...) to eventually do it.

TLDR: main difficulty is that almost all abstractions (TNF, Deciders, ListTape, etc...) depend on `BB52Statement.v` for definition of Sigma and St.

**Comment by ccz181078** [comment]
solved by https://github.com/ccz181078/Coq-BB5/pull/15 .
